Detectors/TPC/simulation/include/TPCSimulation/DigitPad.h

```diff
@@ -78,45 +84,76 @@ class DigitPad{
 
     float                  mChargePad;   ///< Total accumulated charge on that pad for a given time bin
     unsigned char          mPad;         ///< Pad of the ADC value
+#ifdef EXTLABELS
+    unsigned int           mId;          ///< An integer id for this digit (can be combined with mPad to not waste memory??)
+#else
     std::vector<std::pair<MCCompLabel, int>> mMClabel; ///< vector to accumulate the MC labels
-
+#endif
     // TODO: optimize this treatment, for example by using a structure like this
     // struct MCIDValue {
     //   unsigned int eventId : 15; // 32k event Id possible
     //   unsigned int trackId: 17; // 128K tracks possible
     //   unsigned int occurences : 32; // 4G occurrences possible
     // }
     // std::vector<MCID> mMCID;
+
+    // a global memory space where we keep intermediate monte carlo labels
+    // (this avoids having to use a std::vector<> member inside each digit which has at leas 24bytes overhead)
+    static o2::dataformats::LabelContainer<std::pair<MCCompLabel, int>, false> sLabels;
```

Comment:
@a-mathis, @wiechula : This is what I discussed previously in one of our meetings. The new labelcontainer can also serve as a global accumulator of labels and we can remove std::vector members from any data object.

Some problems to solve:
a) can we avoid the mId member or pack it together with mPad?
b) when do we cleanup sLabels? I had a look and it seems that DigitPad::reset is never actually called. Only the destructor of DigitPad

Reply:
a) I have a general question here: the mID I suppose is a global ID for the matching of the labels in the container to a specific DigitPad, right? This would mean the number is a combination of sector, time bin, global pad number (mPad)... As far as I can see, one could either try to make a combination of the relevant figures such that in the end we can again obtain mPad, OR for the time being simply replace mPad with mID (which should at the moment cause no problems whatsoever).
b) A given DigitPad is persistent in memory as long as possibly a correlation with further incoming events (and for that time can accumulate further charges, and hence MC labels) and afterwards the underlying structures (DigitTime) is removed from memory. Therefore, there is from my point of view no need to reset neither the DigitPad, nor the sLabels.